From 2c07266a13357ca7b0867809ad009dcf9ebf5025 Mon Sep 17 00:00:00 2001
From: Li Lingfeng <lilingfeng3@huawei.com>
Date: Wed, 18 Jan 2023 10:58:40 +0800
Subject: [PATCH] fix kabi broken due to may_pollfree
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 2c07266a13357ca7b0867809ad009dcf9ebf5025
Modified-by-SEL: No


hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I685FC
CVE: NA

--------------------------------

Commit 0845c5803f3f("[Backport] io_uring: disable polling pollfree
files") adds a new member in file_operations, so we need to fix
kabi broken problem.

Signed-off-by: Li Lingfeng <lilingfeng3@huawei.com>
Reviewed-by: Zhang Yi <yi.zhang@huawei.com>
Signed-off-by: Jialin Zhang <zhangjialin11@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/fs.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/include/linux/fs.h b/include/linux/fs.h
index cf0faea8ff6d..98236a86cca0 100644
--- a/include/linux/fs.h
+++ b/include/linux/fs.h
@@ -1898,9 +1898,8 @@ struct file_operations {
 				   struct file *file_out, loff_t pos_out,
 				   loff_t len, unsigned int remap_flags);
 	int (*fadvise)(struct file *, loff_t, loff_t, int);
-	bool may_pollfree;
 
-	KABI_RESERVE(1)
+	KABI_USE(1, bool may_pollfree)
 	KABI_RESERVE(2)
 	KABI_RESERVE(3)
 	KABI_RESERVE(4)
-- 
2.33.0

